### PR TITLE
feat: support permission on list_collaborators

### LINKED ISSUE #629: list_collaborators with permission
https://docs.github.com/en/rest/collaborators/collaborators?apiVersion=2022-11-28#list-repository-collaborators

It has a `permission` param. But this doesn't seem to exist at `ListCollaboratorsBuilder`.

### PR DESCRIPTION
**Pull request by tisonkun**
This closes #629.

### CODE REVIEW

File: src/api/repos/collaborators.rs [hunk 1]
```diff
@@ -8,6 +8,8 @@ pub struct ListCollaboratorsBuilder<'octo, 'r> {
     per_page: Option<u8>,
     #[serde(skip_serializing_if = "Option::is_none")]
     page: Option<u32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    permission: Option<String>,
```
Comment (tisonkun) [inline src/api/repos/collaborators.rs:12]: Unsure if we should use an enum for the possible values instead.

I'm not quite familiar with enum serialization.

File: src/api/repos/collaborators.rs [hunk 2]
```diff
@@ -31,6 +33,14 @@ impl<'octo, 'r> ListCollaboratorsBuilder<'octo, 'r> {
         self
     }
 
+    /// Filter collaborators by the permissions they have on the repository.
+    /// If not specified, all collaborators will be returned.
+    /// Can be one of: pull, triage, push, maintain, admin
+    pub fn permission(mut self, page: impl ToString) -> Self {
+        self.page = Some(page.to_string());
```
Comment (tisonkun) [inline src/api/repos/collaborators.rs:40]:
```suggestion
    pub fn permission(mut self, permission: impl ToString) -> Self {
        self.permission = Some(permission.to_string());
```

File: src/api/repos/collaborators.rs (hunk 1 above)
Comment (XAMPPRocky) [reply src/api/repos/collaborators.rs:12]: Is there a known limited set of values it should be? Then it should use an `enum`. You can copy one of the other enums in the crate as an example.

Comment (tisonkun) [reply src/api/repos/collaborators.rs:12]: Sounds reasonable. Let me update the code here in hours.

Comment (tisonkun) [reply src/api/repos/collaborators.rs]: Updated.

**Comment by tisonkun** [comment]
@XAMPPRocky I've run `cargo fmt --all` locally.

**Comment by XAMPPRocky** [comment]
@tisonkun Sorry for the delay, could you rebase this PR?

**Comment by XAMPPRocky** [comment]
Thank you for your PR, and congrats on your first contribution! 🎉

**Pull request merged**